packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -68,21 +68,34 @@ const schemaFields = {
   frequency: z.object({ ...comboSchema }, { invalid_type_error: 'Please select a frequency' }),
 };
 
-const medicationOrderFormSchema = z.discriminatedUnion('isFreeTextDosage', [
-  z.object({
-    ...schemaFields,
-    isFreeTextDosage: z.literal(false),
-    freeTextDosage: z.string().optional(),
-  }),
-  z.object({
-    ...schemaFields,
-    isFreeTextDosage: z.literal(true),
-    dosage: z.number().nullable(),
-    unit: z.object({ ...comboSchema }).nullable(),
-    route: z.object({ ...comboSchema }).nullable(),
-    frequency: z.object({ ...comboSchema }).nullable(),
-  }),
-]);
+const medicationOrderFormSchema = z
+  .discriminatedUnion('isFreeTextDosage', [
+    z.object({
+      ...schemaFields,
+      isFreeTextDosage: z.literal(false),
+      freeTextDosage: z.string().optional(),
+    }),
+    z.object({
+      ...schemaFields,
+      isFreeTextDosage: z.literal(true),
+      dosage: z.number().nullable(),
+      unit: z.object({ ...comboSchema }).nullable(),
+      route: z.object({ ...comboSchema }).nullable(),
+      frequency: z.object({ ...comboSchema }).nullable(),
+    }),
+  ])
+  .refine(
+    (formValues) => {
+      if (formValues.pillsDispensed > 0) {
+        return Boolean(formValues.quantityUnits);
+      }
+      return true;
+    },
+    {
+      message: 'Please select Quantity unit',
```

Comment:
```suggestion
      message: 'Please select a quantity unit',
```

@vasharma05, could you guide @mccarthyaaron on how to make these error messages translatable? We should probably add an example to our [Coding Conventions](https://o3-docs.openmrs.org/docs/coding-conventions).

Reply:
Will be eager to learn how to do that. Thanks

Reply:
Hi @mccarthyaaron!
You can refer #1652.
Thanks!